marshal.go

```diff
@@ -266,6 +266,31 @@ func marshalImpl(vo reflect.Value, w io.Writer, pos uint64, parent *Element, opt
 					}
 					pos, err = writeOne(lst[0])
 				}
+			case reflect.Func:
+				ret := vn.Call(nil)
+				lenRet := len(ret)
+				if lenRet != 1 && lenRet != 2 {
+					return pos, wrapErrorf(ErrIncompatibleType, "Number of return value must be 1 or 2 but %d", lenRet)
```

Comment:
```suggestion
					return pos, wrapErrorf(ErrIncompatibleType, "number of return value must be 1 or 2 but %d", lenRet)
```
https://github.com/golang/go/wiki/Errors#errors
> Error strings should not start with a capital letter because they'll often be prefixed before printing:

Reply:
Ah, I forgot that, fixed.